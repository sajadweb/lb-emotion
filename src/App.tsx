import * as React from 'react';
import {   Navbar } from './components';

class App extends React.Component {
  public render() {
    return (
      <div>
        <Navbar>
          <div> mdsds</div>
        </Navbar>
        {/* <Alert primary >
          To get started, edit <code>src/App.tsx</code> and save to reload.
        </Alert> */}
      </div>
    );
  }
}

export default App;
