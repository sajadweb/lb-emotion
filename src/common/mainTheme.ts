import { Theme } from './styled';
import * as color from 'color';
function darken(hex: string) {
    return color(hex)
        .darken(0.03)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#24292e',
        secondary_dark: darken('#24292e'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#d4edda',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    darken: (_color?: string) => {
        return color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return color(hex).fade(a ? a : 0.5).string();
    }
};

export default mainTheme;
