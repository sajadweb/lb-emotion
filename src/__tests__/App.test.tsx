import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import App from '../App';
import mainTheme from '../common/mainTheme';
import { ThemeProvider } from 'emotion-theming';

test('renders with correct styles', () => {
  const tree = renderer.create(<ThemeProvider theme={mainTheme}>
    <App />
  </ThemeProvider>).toJSON();
  expect(tree).toMatchSnapshot();
});