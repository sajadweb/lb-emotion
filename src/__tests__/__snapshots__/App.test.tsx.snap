// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<div>
  <p>
    To get started, edit 
    <code>
      src/App.tsx
    </code>
     and save to reload.
  </p>
</div>
`;
